### PR TITLE
List of publications

### PR DESCRIPTION
**Pull request by kralka**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,45 +1,57 @@
...
 
-1. Clone the repository: `git clone github.com/google/scaaml/`
-2. Create and activate Python virtual environment:
-      `python3 -m venv my_env`
-      `source my_env/bin/activate`
-3. Install dependencies: `python3 -m pip install --require-hashes -r requirements.txt`
-4. Install the SCAAML package: `python setup.py develop`
+1.  Clone the repository: `git clone github.com/google/scaaml/`
+2.  Create and activate Python virtual environment:
+       `python3 -m venv my_env`
```
Comment (jmichelp) [inline README.md:32]: This is not rendered as 2 separate lines. Is it intended?

File: README.md [hunk 2]
```diff
@@ -1,45 +1,57 @@
...
+`pip install --require-hashes -r base-tooling-requirements.txt`
 
-Update: `pip-compile --allow-unsafe requirements.in --generate-hashes --upgrade` and commit requirements.txt.
+Update: `pip-compile --allow-unsafe requirements.in --generate-hashes --upgrade`
+and commit requirements.txt.
 
 ### Dataset and models
 
-Every SCAAML component rely on a datasets and optional models that you will need to download in the component directory. The link to download those are available in the components specific README.md. Simply click on the directory representing the component of your choice, or the link to the component in the list above.
+Every SCAAML component rely on a datasets and optional models that you will need
```
Comment (jmichelp) [inline README.md:51]: s/datasets/dataset/ (or remove the article if you want to keep plural)

File: README.md [hunk 3]
```diff
@@ -1,45 +1,57 @@
 # SCAAML: Side Channel Attacks Assisted with Machine Learning
+
 ![SCAAML banner](https://storage.googleapis.com/scaaml-public/visuals/scaaml-banner.png)
 
-SCAAML (Side Channel Attacks Assisted with Machine Learning) is a deep learning framework dedicated to side-channel attacks.
-It is written in python and run on top of TensorFlow 2.x.
+SCAAML (Side Channel Attacks Assisted with Machine Learning) is a deep learning
+framework dedicated to side-channel attacks.  It is written in python and run on
```
Comment (jmichelp) [inline README.md:6]: nit: 2 spaces instead of 1

File: README.md [hunk 4]
```diff
@@ -1,45 +1,57 @@
...
-- [`scaaml/`](https://github.com/google/scaaml/tree/master/scaaml/): The SCAAML framework code. Its used by the various tools.
-- [`scaaml_intro/`](https://github.com/google/scaaml/tree/master/scaaml_intro): *A Hacker Guide To Deep Learning Based Side Channel Attacks*.
-  Code, dataset and models used in our step by step tutorial on how to use deep-learning to perform AES side-channel attacks in practice.
+-   [`scaaml/`](https://github.com/google/scaaml/tree/master/scaaml/): The
+    SCAAML framework code. Its used by the various tools.
+
+-   [`scaaml_intro/`](https://github.com/google/scaaml/tree/master/scaaml_intro):
+    *A Hacker Guide To Deep Learning Based Side Channel Attacks*.  Code, dataset
+    and models used in our step by step tutorial on how to use deep-learning to
+    perform AES side-channel attacks in practice.
```
Comment (jmichelp) [inline README.md:19]: Should we put a warning that the scaaml tooling evolved without this demo being maintained?

File: README.md [hunk 5]
```diff
@@ -1,45 +1,57 @@
...
-4. Install the SCAAML package: `python setup.py develop`
+1.  Clone the repository: `git clone github.com/google/scaaml/`
+2.  Create and activate Python virtual environment:
+       `python3 -m venv my_env`
+       `source my_env/bin/activate`
+3.  Install dependencies: `python3 -m pip install --require-hashes -r
+    requirements.txt`
+4.  Install the SCAAML package: `python setup.py develop`
 
 ### Update dependencies
```
Comment (jmichelp) [inline README.md:38]: I would add a comment (or move this to a dedicated "Maintainer" section) that this is for maintainers only.
I.e. we don't expect end users to do it

File: README.md [hunk 6]
```diff
@@ -1,45 +1,57 @@
...
 
-Update: `pip-compile --allow-unsafe requirements.in --generate-hashes --upgrade` and commit requirements.txt.
+Update: `pip-compile --allow-unsafe requirements.in --generate-hashes --upgrade`
+and commit requirements.txt.
 
 ### Dataset and models
 
-Every SCAAML component rely on a datasets and optional models that you will need to download in the component directory. The link to download those are available in the components specific README.md. Simply click on the directory representing the component of your choice, or the link to the component in the list above.
+Every SCAAML component rely on a datasets and optional models that you will need
+to download in the component directory. The link to download those are available
```
Comment (jmichelp) [inline README.md:52]: spell: download these is

File: README.md [hunk 7]
```diff
@@ -56,21 +68,14 @@ its codebase, models or datasets please cite:
...
-@misc{bursztein2023generic,
-      title={Generic Attacks against Cryptographic Hardware through Long-Range Deep Learning}, 
-      author={Elie Bursztein and Luca Invernizzi and Karel Král and Daniel Moghimi and Jean-Michel Picod and Marina Zhang},
-      year={2023},
-      eprint={2306.07249},
-      archivePrefix={arXiv},
-      primaryClass={cs.CR}
-}
-```
+### List of all papers
```
Comment (jmichelp) [inline README.md:71]: s/all/our/ ?

File: papers/README.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# SCAAML list of papers and publications
+
+This directory provides the list of papers, presentations, code, and datasets
+used in the publications.
```
Comment (jmichelp) [inline papers/README.md:4]: nit: our publications?

File: papers/2024/GPAM/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+| ECC CM3 | 17,5M        | 122,880           | 8,192 | 8,192   | 3.7                   | [ECC CM3](https://github.com/google/scaaml/tree/main/papers/datasets/ECC/GPAM) |
+
+### Publicly available datasets
+
+[REASSURE (H2020 731591) ECC Dataset](https://zenodo.org/records/3609789)
+Łukasz Chmielewski
+
+[ASCADv2](https://www.data.gouv.fr/en/datasets/ascadv2/)
+Loïc Masure and Rémi Strullu.
+Side-channel analysis against ANSSI’s protected AES implementation on ARM:
```
Comment (jmichelp) [inline papers/2024/GPAM/README.md:45]: I would add a line break here to make it easier to read.
1st line: dataset and authors
2nd line: description and link

WDYT?

File: papers/2024/GPAM/README.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+[ASCADv1](https://github.com/ANSSI-FR/ASCAD)
+Ryad Benadjila, Prouff Emmanuel, and Junwei Wang.
+ATMEGA boolean masked AES variable key
+
+[CHES 2023 SMAesH challenge](https://smaesh-challenge.simple-crypto.org/)
+Gaëtan Cassiers, Charles Momin, and François-Xavier Standaert.
+
+## Models
+
+TODO
```
Comment (jmichelp) [inline papers/2024/GPAM/README.md:58]: maybe "Coming soon..." instead? :)

File: scaaml_intro/README.md [hunk 11]
```diff
@@ -1,18 +1,20 @@
 # SCAAML AES side-channel attacks tutorial
 
-This directory provide the code, models and dataset needed to reproduce the AES deep-learning side channel attack
-demonstrated at [DEF CON 27](https://elie.net/talk/a-hackerguide-to-deep-learning-based-side-channel-attacks/)
+This directory provide the code, models and dataset needed to reproduce the AES
```
Comment (jmichelp) [inline scaaml_intro/README.md:3]: spell: provides
spell: and the dataset (unless there is more than 1)

File: scaaml_intro/README.md [hunk 12]
```diff
@@ -39,29 +41,35 @@ architecture which is significantly more efficient.
...
 | [models.zip](https://storage.googleapis.com/scaaml-public/scaaml_intro/models.zip)     | TinyAES 48 pretrained models - 3 attack points * 16 bytes |     312MB     | `models/`         | 17d7d32cca0ac0db157ae1f5696f6c64bba6d753a8f33802d0d9614bb07d3d9b |
 | [logs.zip](https://storage.googleapis.com/scaaml-public/scaaml_intro/logs.zip)         | Tensorboard training logs (optional)                      |     616MB     | `logs`            | 5b2f43f89990653d64820cca61f15fc6818ee674ae4cc2b4f235cfd9a48f3b28 |
 
-Make sure to unzip them in this directory (`scaaml_demo`) otherwise the code won't find them.
+Make sure to unzip them in this directory (`scaaml_demo`) otherwise the code
+won't find them.
 
-Note: the Tensorboard logs are optional, they are mostly provided for people interested in looking at how fast the models converged.
+Note: the Tensorboard logs are optional, they are mostly provided for people
+interested in looking at how fast the models converged.
```
Comment (jmichelp) [inline scaaml_intro/README.md:62]: just checking: 1 model or more than one?

File: scaaml_intro/README.md [hunk 13]
```diff
@@ -39,29 +41,35 @@ architecture which is significantly more efficient.
...
 
 The code is split into two parts:
 
-- `train.py` is used to train the attack models. It takes as argument a config
-  that define what to train. The configuration used in the tutorial is located here
-  `config/stm32f415_tinyaes.json` and you can use it to train your own models by running
-  `python train.py -c config/stm32f415_tinyaes.json`
-- `key_recovery_demo.ipynb` is the notebook that showcases how to use the trained model
-  and the `scaaml` library to recover TinyAES keys with just 4 traces. You can either use the models provided, or train your own.
+-   `train.py` is used to train the attack models. It takes as argument a config
```
Comment (jmichelp) [inline scaaml_intro/README.md:68]: same as above: how many models?

File: scaaml_intro/README.md [hunk 14]
```diff
@@ -39,29 +41,35 @@ architecture which is significantly more efficient.
...
+    that define what to train. The configuration used in the tutorial is located
```
Comment (jmichelp) [inline scaaml_intro/README.md:69]: spell: defines

File: scaaml_intro/README.md [hunk 15]
```diff
@@ -39,29 +41,35 @@ architecture which is significantly more efficient.
...
+    here `config/stm32f415_tinyaes.json` and you can use it to train your own
+    models by running `python train.py -c config/stm32f415_tinyaes.json`
+
+-   `key_recovery_demo.ipynb` is the notebook that showcases how to use the
+    trained model and the `scaaml` library to recover TinyAES keys with just 4
+    traces. You can either use the models provided, or train your own.
```
Comment (jmichelp) [inline scaaml_intro/README.md:75]: spell: provided models

File: papers/datasets/ECC/GPAM/README.md [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+
+Additive blinding using a 256-bit long random mask.
+
+## ECC CM2
+
+Multiplicative blinding using a 128-bit long random mask.
+
+## ECC CM3
+
+Combination of CM1 and CM2 (CM1 where each each scalar multiplication in CM1 is
```
Comment (jmichelp) [inline papers/datasets/ECC/GPAM/README.md:68]: spell: too many "each"

File: README.md (hunk 4 above)
Comment (kralka) [reply README.md:19]: Last time I was checking the demo worked. If we deem it helpful I can do an integration test for this (another PR).

File: README.md (hunk 1 above)
Comment (kralka) [reply README.md:32]: Fine by me both ways (ow line too long).

File: scaaml_intro/README.md (hunk 12 above)
Comment (kralka) [reply scaaml_intro/README.md:62]: 3*16 (more than one)

File: papers/2024/GPAM/README.md (hunk 9 above)
Comment (kralka) [reply papers/2024/GPAM/README.md]: Not sure what description you have in mind, mostly that is a paper name. Like this?

**Review by jmichelp — APPROVED** [review]

**Pull request merged**